llm/orchestrator-server/src/main/python/indexing_tools/index_documents.py

```diff
@@ -0,0 +1,225 @@
+        doc.metadata['index_datetime'] = formatted_datetime
+        doc.metadata['id'] = uuid4() # A uuid for the doc (will be used by TOCK)
+
+    logging.debug(f"Split texts in {args['<chunks_size>']} characters-sized chunks")
+    # recursive splitter is used to preserve sentences & paragraphs
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size=int(args['<chunks_size>']))
+    splitted_docs = text_splitter.split_documents(docs)
+    # Add chunk id to each chunk
+    tagged_index = 0
+    len_docs = len(splitted_docs)
+    while tagged_index < len_docs:
+        cur_doc_uuid = splitted_docs[tagged_index].metadata['id']
+        chunks_ctr = 1
+        # Search all chunks with same id
+        while (tagged_index+chunks_ctr)<len_docs and splitted_docs[tagged_index+chunks_ctr].metadata['id'] == cur_doc_uuid:
+            chunks_ctr += 1
+        # Add tag
+        doc_nb = 1
+        for doc in splitted_docs[tagged_index:tagged_index+chunks_ctr]:
+            doc.metadata['chunk'] = f"{doc_nb}/{chunks_ctr}"
+            doc_nb += 1
+        tagged_index = tagged_index + chunks_ctr
+    logging.debug(f"Split {len(docs)} texts in {len(splitted_docs)} chunks")
+
+    logging.debug(f"Get embeddings model from {args['<embeddings_cfg>']} config file")
+    with open(args['<embeddings_cfg>'], 'r') as file:
+        config_dict = json.load(file)
+    # Create settings class according to embeddings provider from config file
+    if config_dict['provider'] == EMProvider.OPEN_AI:
+        em_settings = OpenAIEMSetting(**config_dict)
+    elif config_dict['provider'] == EMProvider.AZURE_OPEN_AI_SERVICE:
+        em_settings =AzureOpenAIEMSetting(**config_dict)
+    
+    # Use embeddings factory from orchestrator
+    em_factory = get_em_factory(em_settings)
+    em_factory.check_embedding_model_setting()
+    embeddings = em_factory.get_embedding_model()
+
+    logging.debug(f"Index chunks in DB")
+    # Use vector store factory from orchestrator
+    vectorstore_factory = get_vector_store_factory(VectorStore.OPEN_SEARCH, 
+                                                   embedding_function=embeddings, 
+                                                   index_name=args['<index_name>'])
+    opensearch_db = vectorstore_factory.get_vector_store()
+    # Index respecting bulk_size (500 is from_documents current default: it is described for clarity only)
+    bulk_size = 500
+    for i in range(0, len(splitted_docs), bulk_size):
+        opensearch_db.add_documents(documents=splitted_docs[i:i+500], 
+                                    bulk_size=bulk_size)
+    
+    # Print statistics
+    duration = datetime.now() - start_time
+    logging.debug(f"Indexed {len(splitted_docs)} chunks in '{args['<index_name>']}' from {len(docs)} lines in '{args['<input_csv>']}' (duration: {duration})")
+
+    # Return session index uuid to main script
+    return session_uuid
+
+
+def index_name_is_valid(index_name):
```

Comment:
Add typing for the function.

```suggestion
def index_name_is_valid(index_name: str) -> bool:
```